xcodeproj/internal/xcodeprojinfo.bzl

```
@@ -256,6 +277,19 @@ def _skip_target(*, target, deps, deps_attrs, transitive_infos):
         ),
     )
 
+def _create_test_envs_depset(*, automatic_target_info, ctx, id, target):
+    test_env = getattr(ctx.rule.attr, automatic_target_info.env, {})
+    raw_run_env = target[RunEnvironmentInfo].environment if RunEnvironmentInfo in target else {}
+
+    # Some keys are not applicable in schemes, we will filter them out here
+    run_env = {}
+    denylist_run_env_keys = ["XCODE_VERSION_OVERRIDE", "XCODE_VERSION"]
+    for key, value in raw_run_env.items():
+        if key not in denylist_run_env_keys:
+            run_env[key] = value
+
+    return struct(id = id, env = struct(**dicts.add(test_env, run_env)))
```

We want test_env to override run_env, right?
```suggestion
    return struct(id = id, env = struct(**dicts.add(run_env, test_env)))
```